runtime/src/bank.rs

```diff
@@ -635,21 +636,34 @@ impl Bank {
     where
         F: Fn(&VoteState) -> bool,
     {
-        self.accounts
-            .accounts_db
-            .read()
-            .unwrap()
-            .accounts
-            .values()
-            .filter_map(|account| {
-                if vote_program::check_id(&account.owner) {
-                    if let Ok(vote_state) = VoteState::deserialize(&account.userdata) {
-                        if cond(&vote_state) {
-                            return Some(vote_state);
+        let parents = self.parents();
+        let mut accounts = vec![&self.accounts];
+        accounts.extend(parents.iter().map(|b| &b.accounts));
```

Comment:
don't we already have this code in like 4 places, now? ;)

Reply:
hehe yeah, it got stolen and eaten by bigger badder PRs xD

Reply:
Do I win for cleanest?

```rust
        let parents = self.parents();
        let mut banks = vec![self];
        banks.extend(parents.iter().map(|x| x.as_ref()));
```

That first line pins down the lifetime, which allows the `as_ref()` to get past the borrow-checker.

Reply:
You'd have to break it out as a macro, not a method, though. I'm not aware of a way to returned borrowed refs from Arcs.

Reply:
But why not `.map(|p|` for parent..?